tests/components/zwave/test_node_entity.py

```diff
@@ -164,6 +164,73 @@ def listener(event):
     assert events[0].data[const.ATTR_SCENE_DATA] == scene_data
 
 
+async def test_application_version(hass, mock_openzwave):
+    """Test application version."""
+    mock_receivers = {}
+
+    signal_mocks = [
+        mock_zwave.MockNetwork.SIGNAL_VALUE_CHANGED,
+        mock_zwave.MockNetwork.SIGNAL_VALUE_ADDED,
+    ]
+
+    def mock_connect(receiver, signal, *args, **kwargs):
+        if signal in signal_mocks:
+            mock_receivers[signal] = receiver
+
+    node = mock_zwave.MockNode(node_id=11)
+
+    with patch("pydispatch.dispatcher.connect", new=mock_connect):
+        entity = node_entity.ZWaveNodeEntity(node, mock_openzwave)
+
+    for signal_mock in signal_mocks:
+        assert signal_mock in mock_receivers.keys()
+
+    events = []
+
+    def listener(event):
+        events.append(event)
+
+    # Make sure application version isn't set before
+    assert (
+        node_entity.ATTR_APPLICATION_VERSION
+        not in entity.device_state_attributes.keys()
```

Comment:
It's better to assert the state and state attributes than the entity properties. The state is the truth of the entity. Avoiding interacting directly with the entity in tests makes the tests more robust to changes in the entity implementation.

As this is an existing pattern in the zwave tests, I think we can merge this as is.